src/bridges/notional/NotionalBridge.sol

```diff
@@ -0,0 +1,207 @@
+        }
+        if (
+            _outputAssetA.assetType == AztecTypes.AztecAssetType.ERC20 &&
+            !preapproved[_outputAssetA.erc20Address][ROLLUP_PROCESSOR]
+        ) {
+            // if it's not preapproved, we need to approve it
+            ERC20(_outputAssetA.erc20Address).approve(ROLLUP_PROCESSOR, type(uint256).max);
+            preapproved[_outputAssetA.erc20Address][ROLLUP_PROCESSOR] = true;
+        } else if (_outputAssetA.assetType == AztecTypes.AztecAssetType.ETH) {
+            IRollupProcessor(ROLLUP_PROCESSOR).receiveEthFromBridge{value: outputValueA}(_interactionNonce);
+        }
+    }
+
+    /**
+     * @notice Helper functon to compute the fcash amount
+     * @dev If the token is a ctoken, then we need to convert the amount to its underlying in 8 decimals
+     * @dev If the token is an underlying token, then we simply scale it to 8 decimals
+     * @param _token is the address of input asset token
+     * @param _amount is the amount of input asset token
+     * @return underlyingAmount The amount of underlying token with 8 decimal
+     */
+    function computeUnderlyingAmount(
+        address _token,
+        address _underlyingToken,
+        uint256 _amount
+    ) public returns (uint88) {
+        uint256 underlyingDecimals = ERC20(_underlyingToken).decimals();
+        if (_token != _underlyingToken) {
+            uint256 exchangeRateCurrent = CToken(_token).exchangeRateCurrent();
+            uint256 mantissa = 10 + underlyingDecimals;
+            uint256 underlyingAmount = (_amount * exchangeRateCurrent) / (10**mantissa);
+            return uint88(underlyingAmount);
+        }
+        if (underlyingDecimals >= 8) {
+            return uint88(_amount / 10**(underlyingDecimals - 8));
+        }
+        return uint88(_amount * 10**(8 - underlyingDecimals));
+    }
+
+    /**
+     * @notice Approve the fcash contract to use the input token and mint fcash token
+     * @dev If the input token is ETH, we need to deposit it into WETH. If the token is a ctoken,
+     * we need to call mintViaAsset. Otherwise, we just need to call mintViaUnderlying.
+     * @dev Will revert if the wrapped fcash contract has not been deployed
+     * @param _inputToken The address of the input asset
+     * @param _outputToken The address of the output asset
+     * @param _currencyId The predefined Notional currency id defined in constructor
+     * @param _amount The amount of input token
+     * @param _maturity The maturity of the market we want to enter
+     * @return outputAmount The amount of fcash token we actually receive
+     */
+    function _enter(
+        address _inputToken,
+        address _outputToken,
+        uint16 _currencyId,
+        uint256 _amount,
+        uint40 _maturity
+    ) internal returns (uint256) {
+        IWrappedfCash fcash = IWrappedfCash(FCASH_FACTORY.computeAddress(_currencyId, _maturity));
```

Comment:
Please remove this check. Once the approval will be only in the separate "approvals" function this method would revert anyway in case there would be inocorrect token. With this in mind this check only wastes gas.